### PR TITLE
Update icms13 installer for php7

### PR DESCRIPTION
**Pull request by fiammybe**
This should get the 1.3 branch working on PHP7. I propose naming this branch ImpressCMS 1.4 to indicate the major change (PHP7 compatibility).

### CODE REVIEW

File: htdocs/include/common.php [hunk 1]
```diff
@@ -38,8 +38,12 @@
 /** make sure mainfile is included, for security and functionality */
 defined("XOOPS_MAINFILE_INCLUDED") or die();
 
-/** @todo This funtion has been deprecated in PHP 5.3.0 - need to adjust in 1.4 ~skenow */
-@set_magic_quotes_runtime(0);
+/** @todo when this is no longer possible to run under PHP 5.x, we can remove the check - fiammybe 12 may 2019 */
```
Comment (MekDrop) [inline htdocs/include/common.php]: I think we can remove that. PHP 5.x are now unsupported versions. We should not care about this anymore.

File: htdocs/install/page_dbconnection.php [hunk 2]
```diff
@@ -45,37 +45,59 @@
 }
 
 $error = '';
-if ($_SERVER['REQUEST_METHOD'] == 'POST' && !empty( $vars['DB_HOST'] ) && !empty( $vars['DB_USER'] )) {
-	$func_connect = empty( $vars['DB_PCONNECT'] ) ? "mysql_connect" : "mysql_pconnect";
-	if (! ( $link = @$func_connect( $vars['DB_HOST'], $vars['DB_USER'], $vars['DB_PASS'], true ) )) {
-		$error = ERR_NO_DBCONNECTION;
+if ($_SERVER['REQUEST_METHOD'] == 'POST' && !empty($vars['DB_HOST']) && !empty($vars['DB_USER'])) {
+	switch ($vars['DB_TYPE']) {
+		case 'mysql':
```
Comment (MekDrop) [inline htdocs/install/page_dbconnection.php]: again. should we care about old mysql drivers that doesn't work on unsupported PHP version?

File: htdocs/install/page_dbconnection.php [hunk 3]
```diff
@@ -45,37 +45,59 @@
...
 }
 
-if (@empty( $vars['DB_HOST'] )) {
-	//so far, mysql extenstion has to exist and be loaded
-	$connections = array(
-			'mysql' => array('type' => 'mysql', 'name' => 'MySQL', 'selected' => 'selected'),
-	);
+//so far, mysql extension has to exist and be loaded
+$connections = [];
+if (function_exists('mysql_connect') || function_exists('mysql_pconnect')) {
```
Comment (MekDrop) [inline htdocs/install/page_dbconnection.php]: same

File: htdocs/install/page_dbconnection.php [hunk 4]
```diff
@@ -45,37 +45,59 @@
...
+	$connections['mysql'] = array('type' => 'mysql', 'name' => 'MySQL', 'selected' => 'selected');
 	$db_connection = $connections['mysql'];
-	// Fill with default values
-	// check for PDO MySQL and select it, if it is available
-	if (class_exists("PDO", FALSE)) {
-		$db_connection = array('type' => 'pdo.mysql', 'name' => 'PDO MySQL', 'selected' => 'selected');
+}
+// Fill with default values
+// check for PDO MySQL and select it, if it is available
+if (class_exists("PDO", false)) {
```
Comment (MekDrop) [inline htdocs/install/page_dbconnection.php]: better add to required php extensions check. PDO should be required not optional.

File: htdocs/install/page_dbsettings.php [hunk 5]
```diff
@@ -13,54 +13,98 @@
...
 $pageHasHelp = true;
 
 $vars = & $_SESSION ['settings'];
 
-$func_connect = empty ( $vars ['DB_PCONNECT'] ) ? "mysql_connect" : "mysql_pconnect";
-if (! ($link = @$func_connect ( $vars ['DB_HOST'], $vars ['DB_USER'], $vars ['DB_PASS'], true ))) {
-	$error = ERR_NO_DBCONNECTION;
-	$wizard->redirectToPage ( '-1', $error );
+switch ($vars['DB_TYPE']) {
+	case 'mysql':
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php]: again

File: htdocs/install/page_dbsettings.php [hunk 6]
```diff
@@ -13,54 +13,98 @@
...
-	foreach ( $keys as $k) {
-		$vars [$k] = defined ( "XOOPS_$k" ) ? constant ( "XOOPS_$k" ) : '';
+if (!isset ($vars ['DB_NAME']) || false !== @strpos($_SERVER ['HTTP_CACHE_CONTROL'], 'max-age=0')) {
+	$keys = array('DB_NAME', 'DB_CHARSET', 'DB_COLLATION', 'DB_PREFIX', 'DB_SALT');
+	foreach ($keys as $k) {
+		$vars [$k] = defined("XOOPS_$k")? constant ("XOOPS_$k"):'';
+	}
+}
+
+function exec_query($sql, $link) {
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php]: only leave here PDO code

File: htdocs/install/page_dbsettings.php [hunk 7]
```diff
@@ -13,54 +13,98 @@
...
+	if ($link instanceof PDO) {
+		return $link->query($sql);
+	} else {
+		return mysql_query($sql, $link);
+	}
+}
+
+function fetch_assoc($result) {
+	if ($result instanceof PDOStatement) {
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php]: only leave here PDO code

File: htdocs/install/page_dbsettings.php [hunk 8]
```diff
@@ -13,54 +13,98 @@
...
+		return $result->fetch(PDO::FETCH_ASSOC);
+	} else {
+		return mysql_fetch_assoc($result);
+	}
+}
+
+function quote_sql($sql) {
+	global $link;
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php]: only leave here PDO code

File: htdocs/install/page_dbsettings.php [hunk 9]
```diff
@@ -87,40 +131,47 @@ function getDbCollations($link, $charset) {
...
+		if (!isset ($collations [$collation])) {
+			$error = sprintf(ERR_INVALID_DBCOLLATION, $collation);
 		}
 	}
 
 	return $error;
 }
 
+function getDBVersion($link) {
+	if ($link instanceof PDO) {
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php]: only leave here PDO code

File: htdocs/install/page_dbsettings.php [hunk 10]
```diff
@@ -148,87 +199,99 @@ function xoFormFieldCollation($name, $value, $label, $help = '', $link, $charset
...
+	$block .= xoFormFieldCollation($name, $value, $label, $help, $link, $charset);
 	$block .= '</div>';
 
 	return $block;
 }
 
-if ($_SERVER ['REQUEST_METHOD'] == 'GET' && isset ( $_GET ['charset'] ) && @$_GET ['action'] == 'updateCollation') {
-	echo xoFormFieldCollation ( 'DB_COLLATION', $vars ['DB_COLLATION'], DB_COLLATION_LABEL, DB_COLLATION_HELP, $link, $_GET ['charset'] );
+function select_db($db_name, $link) {
+	if ($link instanceof PDO) {
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php]: only leave here PDO code

**Comment by fiammybe** [comment]
I understand your concerns, @MekDrop but shouldn't we have at last 1 version that supports both the old environment and the new? That way people can upgrade their site to the 1.4.0 release, check that everything works, and then upgrade their hosting.

**Comment by MekDrop** [comment]
Ok than maybe alternative solution - popup or block in admin if user is using pre PHP 7.1 version. We somehow need to force our users leave 5.x ASAP and use still supported PHP version.

**Comment by skenow** [comment]
The installer would not affect any existing sites. Handling an upgrade requires different considerations - the current setting in mainfile.php, especially.

**Pull request merged**